### ISSUE TITLE
Leverage "Provides:" to support multiple providers for a given service

### ISSUE THREAD

**Issue by thac0** [clearlinux/distribution#629]
**Is your feature request related to a problem? Please describe.**
Tl;dr - yes. As it stands, if the *desktop* bundle is installed, both *systemd-networkd* and *NetworkManager* end up running side by side. Rather than having NetworkManager *replace* it, NetworkManager is crippled in Clear such that it only manages non-ethernet interfaces. That is a hack, not a solution.

**Describe the solution you'd like**
A robust **Provides:** system is, where packages that provide, say, network interface management can be selected from.

**Describe alternatives you've considered**
I haven't. Other mature distributions have a notion of a "provider", where multiple packages can "provide" a given service. It is then left to the end-user to decide which package will provide said service.

**Additional context**
N/A

**Comment by ahkok** [comment]
@busykai is working on a set of changes that will let NM manage all interfaces. This will largely solve that problem.

On the bundle side of things, we will introduce something like `weak` dependencies.

However, the base guideline that bundles may not conflict (and must be able to be present on a system together without breaking anything) will still be applied, so we're looking at other changes to solve that problem - mostly by making software live together without crippling one or another.
